docs/decisions/0003-learning-path-enrollment-api.rst

```diff
@@ -0,0 +1,125 @@
+0003 Learning Path Enrollment API
+###########################################
+
+Status
+******
+
+**Draft**
+
+.. Standard statuses
+    - **Draft** if the decision is newly proposed and in active discussion
+    - **Provisional** if the decision is still preliminary and in experimental phase
+    - **Accepted** *(date)* once it is agreed upon
+    - **Superseded** *(date)* with a reference to its replacement if a later ADR changes or reverses the decision
+
+    If an ADR has Draft status and the PR is under review, you can either use the intended final status (e.g. Provisional, Accepted, etc.), or you can clarify both the current and intended status using something like the following: "Draft (=> Provisional)". Either of these options is especially useful if the merged status is not intended to be Accepted.
+
+Context
+*******
+
+Learner enrollments in Learning Paths are captured in the
+LearningPathEnrollment model. However, there is no API exposing an interface to
+perform operations on the model.
+
+Decision
+********
+
+We will create the following views in the ``api`` app to expose an API that can be called
+by other services and MFEs.
+
+.. _enroll-api:
+
+1. Enroll API
+=============
+
+This API will allow learners to be enrolled in the
+
++---------------------+-------------------------------------------------------+
+| API Path            | /api/v1/learning-path-enrollment/                     |
++---------------------+-------------------------------------------------------+
+| Methods             | GET, POST, DELETE                                     |
++---------------------+-------------------------------------------------------+
+| Data                | learning_path_id (uuid), username (OPTIONAL)          |
++---------------------+-------------------------------------------------------+
+| Permissions Required| LoggedIn                                              |
++---------------------+-------------------------------------------------------+
+
+.. note:: The UUID of the learning_path_id will be replaced with a more
+   user-friendly value later.
+
+GET
+"""
+
+**Corresponding User Action:** Fetch my enrollment
+
+* The data fields ``learning_path_id`` and ``username`` will be passed as query
+  parameters.
+* Return the LearningPathEnrollment object if the logged-in user is enrolled in the
+  specified learning path. Otherwise returns a HTTP 404.
+* When the ``username`` is present, returns all the enrollments in a learning
+  path if the ``currentUser`` is a **staff** or **admin** user.
+
+
+POST
```

Comment:
What should happen when the enrollment already exists?

Reply:
Added HTTP 409 for this scenario.